### ISSUE TITLE
Fix bibliographic#bib_items sortable_call_number field

### ISSUE THREAD

**Issue by hackartisan** [pulibrary/bibdata#1126]
child of pulibrary/orangelight#2348

The [`add_locator_call_no` method](https://github.com/pulibrary/marc_liberation/blob/4bb139d15e87fc0b0df1ac3744d1185e30260b67/app/controllers/bibliographic_controller.rb#L234-L249) looks for a holding location "f". That will never be there with the new alma locations. If this field is needed (@kevinreiss ?) We need to re-enable the [call-number controller specs](https://github.com/pulibrary/marc_liberation/blob/4bb139d15e87fc0b0df1ac3744d1185e30260b67/spec/controllers/bibliographic_controller_spec.rb#L191-L219) with new fixtures and make sure it works.

This should happen when the holding location is `firestone#stacks`

Re: https://github.com/pulibrary/firestone_locator/blob/00ad7b643305ea055bee34cd717446aa950bb399/includes/item_proc.php#L141

The call number coming out of the Alma endpoint puts subfield k before the call number, so "oversize" and "folio" (and any other subfield k value) would be before the call number.

We'll need to adjust the call number as it comes from the Alma API to put the word "Oversize" at the end. We can do this by calling Alma's holding API, but it would be better to get it out of solr. In Alma we decide to index the subfield k that could include 'Oversize' at the end of the call_number.
We will use a new 'call_number_locator' field https://github.com/pulibrary/bibdata/issues/1136 for the lookup of the firestone locator when the record is in firestone. This new field will not include the 'Oversize' string.

**Comment by hackartisan** [comment]
The locator has a notion of call number ranges that it uses to pinpoint the spot on the floor plan. The sortable call number is the primary data point for this logic.

**Comment by kevinreiss** [comment]
For context here is what the configuration on the shelf location ranges looks like in the locater app: 
<img width="1311" alt="Screen Shot 2021-03-10 at 12 19 46 PM" src="https://user-images.githubusercontent.com/278825/110987854-52c3c880-8335-11eb-933b-5a650417a80f.png">

After looking over the code again, it appears that treatment of oversize books kicks in when the call number ends in "q", this behavior happens only for the existing "f" holding location which will become firestone$stacks in Alma. 

https://github.com/pulibrary/firestone_locator/blob/00ad7b643305ea055bee34cd717446aa950bb399/includes/item_proc.php#L163

https://github.com/pulibrary/firestone_locator/blob/00ad7b643305ea055bee34cd717446aa950bb399/includes/item_proc.php#L263

**Comment by hectorcorrea** [comment]
Some more details about this issue.

Record `9941598513506421` is currently indexed in Solr with the following three call number fields and values:

```
"call_number_display":["Oversize RA566.27  .B7544 2003q"],
"call_number_browse_s":[" .B7544 2003q Oversize RA566.27"],
"call_number_locator_display":[" .B7544 2003q"],
```

`call_number_locator_display` was added by PR https://github.com/pulibrary/bibdata/pull/1138

BibData localhost/bibliographic/9941598513506421/items returns the following (from the Alma API): 

```
{
  "firestone$stacks":[{
    "holding_id":"22172120500006421",
    "call_number":"Oversize RA566.27 .B7544 2003q",
    "items: [{
      "pid":"23172120490006421",
      "id":"23172120490006421",
      "temp_location":null,
      "perm_location":"firestone$stacks",
      "cdl":false
    }]
  }]
}
```

**Comment by mzelesky** [comment]
Here are the values in subfield c of the 'f' location in Voyager, with their frequency of occurrence:
Oversize -> 30137
G53 -> 1
uncat to John Logan -> 1
2 -> 2
CAS -> 1
Oversize CD1040 -> 1
P767 1999 -> 1
tr fr dixn -> 6
t fr dixn -> 1
Oversize RA566.27 -> 1
c2005 -> 1
Overisize -> 1
Oversize DA990.U46 -> 1
Oversize PE1128.A2 -> 1
Oversize Oversize -> 5
Overszie -> 2
Oversize f -> 1
oversize -> 4
Oversized -> 5
Overszize -> 1
Give to Circ for Morgan Robinson -> 1
Ovrersize -> 2
Overeize -> 1
Oversie -> 1

So it looks like just handling 'Folio' and 'Oversize' should suffice.

**Comment by christinach** [comment]
closed by #1459
